diff --git a/vllm/utils/deep_gemm.py b/vllm/utils/deep_gemm.py
index 2a0d4eee4aba..0eb979ce05a5 100644
--- a/vllm/utils/deep_gemm.py
+++ b/vllm/utils/deep_gemm.py
@@ -112,7 +112,7 @@ def is_deep_gemm_supported() -> bool:
     Currently, only Hopper and Blackwell GPUs are supported.
     """
     is_supported_arch = current_platform.support_deep_gemm()
-    return envs.VLLM_USE_DEEP_GEMM and has_deep_gemm() and is_supported_arch
+    return is_supported_arch and envs.VLLM_USE_DEEP_GEMM and has_deep_gemm()
 
 
 @functools.cache
